packages/trie/src/proof/index.ts

```diff
@@ -64,4 +65,104 @@ export function verifyTrieRangeProof(
   )
 }
 
+/**
+ * Creates a proof from a trie and key that can be verified using {@link verifyTrieProof}. An (EIP-1186)[https://eips.ethereum.org/EIPS/eip-1186] proof contains
+ * the encoded trie nodes from the root node to the leaf node storing state data. The returned proof will be in the format of an array that contains Uint8Arrays of
+ * serialized branch, extension, and/or leaf nodes.
+ * @param key key to create a proof for
+ */
+export async function createProof(trie: Trie, key: Uint8Array): Promise<Proof> {
+  trie['DEBUG'] && trie['debug'](`Creating Proof for Key: ${bytesToHex(key)}`, ['CREATE_PROOF'])
+  const { stack } = await trie.findPath(trie['appliedKey'](key))
+  const p = stack.map((stackElem) => {
+    return stackElem.serialize()
+  })
+  trie['DEBUG'] && trie['debug'](`Proof created with (${stack.length}) nodes`, ['CREATE_PROOF'])
+  return p
+}
+
+/**
+ * Updates a trie from a proof by putting all the nodes in the proof into the trie. If a trie is being updated with multiple proofs, {@param shouldVerifyRoot} can
+ * be passed as false in order to not immediately throw on an unexpected root, so that root verification can happen after all proofs and their nodes have been added.
```

Comment:
This comment seems outdated, mentioning that `{@param shouldVerifyRoot} can be passed as false` seems to imply that it does not default to false (which it does). Imo we should clarify this, and add a mention the the `shouldVerifyRoot` param natspec that it defaults to false (I would naturally assume that root is verified).